### DISCUSSION TITLE
Subscriptions and Pagination

### DISCUSSION THREAD

**Discussion by Miloshinjo** [urql-graphql/urql#1691] [Q&A] [answered]
What is the proposed/best way to handle when you want to subscribe to some data that is paginated? 

Pagination has its own `variables` that are not available to subscription, so when I use `cache.updateQuery` in a sub (to update all entities), I don't have access to all entities, because I can only get a query through her correct arguments. 

I guess my question is, how do you create/update/delete in the cache for subscription data, when you have bunch of queries (paginated) and not only query for your data (eg. todos). 

Is there any example or thread I can be pointed to? :) 

Thanks!

**Comment by JoviDeCroock** [comment, marked as answer]
I don't specifically know what you mean here, a subscription generally is used to notify you about an entity that has changed. For instance we're looking at a paginated-query of todos and the second item has changed, when the subscription message comes in graphcache will attempt to automatically update `Todo:2` if it finds it in cache. If not you can use something like the following:

```js
cache.writeFragment(gql`
  fragment _ on Todo {
    id
    text
  }
`, { id: subscription.data.todo.id, text: subscription.data.todo.text })
```

Do note that if the `__typename` and `id` are present in the subscription that this _should_ happen automatically.

**Reply by Miloshinjo** [reply]
Thank you for the reply, it is on the spot. Maybe for me it doesn't happen because of the types (PaginationResult, nodes: Todos etc....),  so I will have to do it manually. 

But I know where to go now :) thanks
